### PR TITLE
WIP [Federation] Add integration test for configmaps

### PR DESCRIPTION
**Pull request by marun**
**Warning:** This PR depends on #42025 and only the following commits are intended to be reviewed here:

 - [fed: Add integration test for configmaps](https://github.com/kubernetes/kubernetes/pull/43125/commits/69e0164b71f2a5324a845fff6205bff2d13beae1)

cc: @kubernetes/sig-federation-pr-reviews

### CODE REVIEW

**Comment by marun** [comment]
cc: @perotinus

File: test/integration/federation/framework/configmap.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	configmapcontroller.StartConfigMapController(config, f.stopChan, true)
+}
+
+func (f *ConfigMapFixture) TearDown(t *testing.T) {
+	close(f.stopChan)
+}
+
+func (f *ConfigMapFixture) Kind() string {
+	adapter := &ConfigMapAdapter{}
+	return adapter.Kind()
```
Comment (perotinus) [inline test/integration/federation/framework/configmap.go]: Why not f.Adapter().Kind()?

File: test/integration/federation/framework/configmap.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+func (f *ConfigMapFixture) Adapter() ResourceAdapter {
+	return f.adapter
+}
+
+func (f *ConfigMapFixture) NewObject(namespace string) pkgruntime.Object {
+	return NewTestConfigMap(namespace)
+}
+
+func NewTestConfigMap(namespace string) pkgruntime.Object {
```
Comment (perotinus) [inline test/integration/federation/framework/configmap.go]: Any reason to make this public?

Comment (marun) [inline test/integration/federation/framework/configmap.go]: It's intended for use in e2e.  If you prefer I could make it private here and upgrade it to public in the PR that enabled configmap e2e.

File: test/integration/federation/framework/configmap.go (hunk 1 above)
Comment (marun) [inline test/integration/federation/framework/configmap.go]: It's a chicken/egg problem.  ``Kind()`` is used by ``TestFederationCRUD`` to identify the test before Setup() is called to initialize f.Adapter().

File: test/integration/federation/framework/configmap.go (hunk 2 above)
Comment (perotinus) [inline test/integration/federation/framework/configmap.go]: Oh, no, that's fine.

**Comment by perotinus** [comment]
The patches LGTM.

**Comment by marun** [comment]
I'm leaving this as WIP because it may make sense to do the following tasks per resource type in a single PR:

 - add type adapter
 - enable integration testing
 - enable upgrade testing
 - enable e2e testing
 - implement the controller with FederationSyncController

However, there are 4 outstanding PRs that need to merge first.

**Comment by marun** [comment]
Hmmm, integration test is running fine locally but failing in CI.

**Comment by marun** [comment]
Closing. @perotinus is going to rework and post a new PR so that configmaps use the new sync controller as well as adding an integration test.

**Pull request closed without merging**